extension/sigv4authextension/extension_test.go

```diff
@@ -0,0 +1,276 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sigv4authextension
```

Comment:
Many of the tests look like they pass locally but would fail on CI due to lack of AWS creds.

Where possible, inject creds from the test (my above suggestion of storing creds to Config may help). For the highest level test where it's not possible, you'll need to SetEnv

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/awsproxy/factory_test.go#L73